src/components/pages/admin/contacts/RoleEditorItem.vue

```diff
@@ -0,0 +1,158 @@
+      </strong>
+      <span>{{ formatLocale(role.dateAdded, 'P') + ' → ' }}</span>
+      {{
+        role.dateExpires
+          ? formatLocale(role.dateExpires, 'P')
+          : t('contacts.data.rolesCopy.today')
+      }}
+      <div class="ml-auto">
+        <AppButton variant="text" size="sm" @click="formVisible = true"
+          >Edit</AppButton
+        >
+        <AppButton variant="dangerText" size="sm" @click="handleDeleteRole">
+          Delete
+        </AppButton>
+      </div>
+    </div>
+
+    <div v-if="formVisible" class="flex flex-1 p-4">
+      <form @submit.prevent="handleFormSubmit" class="flex-initial">
+        <div>
+          <div class="my-2 py-1">
+            <AppLabel :label="inputT('starts.label')" />
+            <div class="flex gap-2">
+              <div>
+                <AppInput v-model="editRole.startDate" type="date" />
+              </div>
+              <div>
+                <AppInput v-model="editRole.startTime" type="time" />
+              </div>
+            </div>
+          </div>
+
+          <div class="my-2 py-1">
+            <AppLabel :label="inputT('expires.label')" />
+            <div class="flex gap-2">
+              <div>
+                <AppInput v-model="editRole.endDate" type="date" />
+              </div>
+              <div>
+                <AppInput v-model="editRole.endTime" type="time" />
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div class="my-3 flex py-3">
+          <AppButton type="submit" variant="primary" :loading="loading">{{
+            t('form.save')
+          }}</AppButton>
+          <AppButton variant="text" class="ml-2" @click="formVisible = false">{{
+            t('form.cancel')
+          }}</AppButton>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import AppLabel from '../../../../components/forms/AppLabel.vue';
+import AppInput from '../../../../components/forms/AppInput.vue';
+import AppButton from '../../../../components/forms/AppButton.vue';
+import AppSelect from '../../../../components/forms/AppSelect.vue';
+import AppRadioGroup from '../../../../components/forms/AppRadioGroup.vue';
+import AppRoundBadge from '../../../../components/AppRoundBadge.vue';
+import { MemberRoleData } from '../../../../utils/api/api.interface';
+import { updateRole, deleteRole } from '../../../../utils/api/member';
+import { onBeforeMount, ref, reactive } from 'vue';
+import { computed } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { formatLocale } from '../../../../utils/dates/locale-date-formats';
+
+const emit = defineEmits(['update:modelValue']);
+const props = defineProps<{
+  role: MemberRoleData;
+  contact: GetMemberData;
+}>();
+const editRole = reactive({
+  role: '',
+  startDate: '',
+  startTime: '',
+  endDate: '',
+  endTime: '',
+});
+
+const { t, n } = useI18n();
+const inputT = (key: string) => t('contacts.data.rolesCopy.' + key);
```

Comment:
Generally prefer using the full `t(..)` unless it's really being used a lot as keeping full i18n keypaths means the keys are more greppable